### PR TITLE
Add EventName to NotifyEventArgs

### LINKED ISSUE #265: Add event name in NotifyEventArgs
Apply https://github.com/neo-project/neo/pull/1701

- Update Nep5Tracker plugin
   - Add `eventName` field in NotifyEventArgs

### LINKED ISSUE #266: An exception in RpcNep5Tracker plugin
**Describe the bug**
An exception in `RpcNep5Tracker` plugin when transfer `gas` or `neo`.

Details:
```
System.Text.DecoderFallbackException
  HResult=0x80070057
  Message=Unable to translate bytes [FA] at index 0 from specified code page to Unicode.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Text.DecoderExceptionFallbackBuffer.Throw(Byte[] bytesUnknown, Int32 index)
   at System.Text.DecoderExceptionFallbackBuffer.Fallback(Byte[] bytesUnknown, Int32 index)
   at System.Text.Encoding.GetCharCountWithFallback(ReadOnlySpan`1 bytes, Int32 originalBytesLength, DecoderNLS decoder)
   at System.Text.Encoding.GetCharCountWithFallback(Byte* pBytesOriginal, Int32 originalByteCount, Int32 bytesConsumedSoFar)
   at System.Text.UTF8Encoding.GetCharCount(Byte* bytes, Int32 count)
   at System.String.CreateStringFromEncoding(Byte* bytes, Int32 byteLength, Encoding encoding)
   at System.Text.Encoding.GetString(ReadOnlySpan`1 bytes)
   at Neo.VM.Helper.GetString(StackItem item) in D:\neo_code\neo\src\neo\VM\Helper.cs:line 201
   at Neo.Plugins.RpcNep5Tracker.HandleNotification(StoreView snapshot, IVerifiable scriptContainer, UInt160 scriptHash, Array stateItems, Dictionary`2 nep5BalancesChanged, UInt16& transferIndex) in D:\neo_code\neo-modules\src\RpcNep5Tracker\RpcNep5Tracker.cs:line 110
   at Neo.Plugins.RpcNep5Tracker.OnPersist(StoreView snapshot, IReadOnlyList`1 applicationExecutedList) in D:\neo_code\neo-modules\src\RpcNep5Tracker\RpcNep5Tracker.cs:line 165
   at Neo.Ledger.Blockchain.Persist(Block block) in D:\neo_code\neo\src\neo\Ledger\Blockchain.cs:line 449
   at Neo.Ledger.Blockchain.OnNewBlock(Block block) in D:\neo_code\neo\src\neo\Ledger\Blockchain.cs:line 336
   at Neo.Ledger.Blockchain.OnInventory(IInventory inventory, Boolean relay) in D:\neo_code\neo\src\neo\Ledger\Blockchain.cs:line 303
   at Neo.Ledger.Blockchain.OnReceive(Object message) in D:\neo_code\neo\src\neo\Ledger\Blockchain.cs:line 376
   at Akka.Actor.UntypedActor.Receive(Object message)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Actor.ActorCell.ReceiveMessage(Object message)
   at Akka.Actor.ActorCell.Invoke(Envelope envelope)
```

**To Reproduce**
1. Deploy a neo node and included the `RpcNep5Tracker` plugin;
2. Send a gas transfer transaction;
3. Wait for transaction confirmation and block persist.

**Screenshots**

image

**Platform:**
 - OS: Windows 10 x64
 - Version: latest version of master (neo, neo-vm, neo-modules, neo-node)

### PR DESCRIPTION
**Pull request by joeqian10**
Close #265, potentially close #266

### CODE REVIEW

**Comment by ProDog** [comment]
I have tested it with https://github.com/neo-project/neo-devpack-dotnet/pull/302, it's working.

My test contract:
```C#
        [DisplayName("TestEvent")]
        public static event Action<byte[], byte[], BigInteger> OnEvent;

        public static bool NotifyTest()
        {            
            OnEvent(new byte[] { 0x89, 0x77, 0x20, 0xd8}, new byte[] { 0x23, 0x8c }, 1);        
            Runtime.Notify("EventName", 11, 12, 13);
            Runtime.Log("test");
            return true;
        }
```

ApplicationLog:
```
"notifications": [
            {
                "contract": "0xa57c54e5c54f82eb3a13591377928274f6bc6573",
                "event_name": "TestEvent",
                "state": {
                    "type": "Array",
                    "value": [
                        {
                            "type": "Buffer",
                            "value": "iXcg2A=="
                        },
                        {
                            "type": "Buffer",
                            "value": "I4w="
                        },
                        {
                            "type": "Integer",
                            "value": "1"
                        }
                    ]
                }
            },
            {
                "contract": "0xa57c54e5c54f82eb3a13591377928274f6bc6573",
                "event_name": "EventName",
                "state": {
                    "type": "Array",
                    "value": [
                        {
                            "type": "Integer",
                            "value": "11"
                        },
                        {
                            "type": "Integer",
                            "value": "12"
                        },
                        {
                            "type": "Integer",
                            "value": "13"
                        }
                    ]
                }
            }
        ]
```

And nep5 transfer notify:
```
"notifications": [
            {
                "contract": "0x668e0c1f9d7b70a99dd9e06eadd4c784d641afbc",
                "event_name": "Transfer",
                "state": {
                    "type": "Array",
                    "value": [
                        {
                            "type": "ByteString",
                            "value": "HANXRkt3fs9rXzrDiTrOH4sWIfY="
                        },
                        {
                            "type": "ByteString",
                            "value": "+pU2/Hks6bMS9XhEc3F6o2fineE="
                        },
                        {
                            "type": "Integer",
                            "value": "123222000"
                        }
                    ]
                }
            }
        ]
```

File: src/ApplicationLogs/LogReader.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ public void OnPersist(StoreView snapshot, IReadOnlyList<Blockchain.ApplicationEx
                 {
                     JObject notification = new JObject();
                     notification["contract"] = q.ScriptHash.ToString();
+                    notification["event_name"] = q.EventName;
```
[the remark this answers is not in the document]
Comment (joeqian10) [reply src/ApplicationLogs/LogReader.cs:67]: "event_name" is consistent with "gas_consumed" and so on:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "sent": [],
        "received": [
            {
                "timestamp": 1579170709527,
                "asset_hash": "0x9bde8f209c88dd0e7ca3bf0af0f476cdd8207789",
                "transfer_address": "NPvKVTGZapmFWABLsyvfreuqn73jCjJtN1",
                "amount": "100",
                "block_index": 54496,
                "transfer_notify_index": 0,
                "tx_hash": "0x49f40ea86de140849fc5eb1c1ab8dbbe4d1186c47807963a162338d979e79c0f"
            },
            {
                "timestamp": 1579170725318,
                "asset_hash": "0x8c23f196d8a1bfd103a9dcb1f9ccf0c611377d3b",
                "transfer_address": "NPvKVTGZapmFWABLsyvfreuqn73jCjJtN1",
                "amount": "10000000000",
                "block_index": 54499,
                "transfer_notify_index": 0,
                "tx_hash": "0x1c25607fda68a2ab5793fb83b5bc87f781afb310127b440620b4ad176d77fa3d"
            }
        ],
        "address": "NNSri1QcdtidykMxryz1xpmzSFwEXeYohH"
    }
}
```

[the remark this answers is not in the document]
Comment (Tommo-L) [reply src/ApplicationLogs/LogReader.cs:67]: ok.

[the remark this answers is not in the document]
Comment (shargon) [reply src/ApplicationLogs/LogReader.cs:67]: We need to unify the naming, we have all kinds

File: src/ApplicationLogs/ApplicationLogs.csproj [hunk 2]
```diff
@@ -6,16 +6,16 @@
     <RootNamespace>Neo.Plugins</RootNamespace>
   </PropertyGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\LevelDBStore\LevelDBStore.csproj" />
+    <ProjectReference Include="..\RpcServer\RpcServer.csproj" />
+  </ItemGroup>
```
Comment (erikzhang) [inline src/ApplicationLogs/ApplicationLogs.csproj:12]: Why modify this file?

Comment (joeqian10) [reply src/ApplicationLogs/ApplicationLogs.csproj:12]: I re-referenced these two projects manually, so the sequence changed. It was an unnecessary change. I'll fix this.

**Comment by ProDog** [comment]
plz merge this, I can test the later PR.

**Pull request merged**